### DISCUSSION TITLE
Unable to install geoviews and cartopy

### DISCUSSION THREAD

**Discussion by ZimmZone** [earthlab-education/Earth-Analytics-AY24#571] [Q&A] [answered]
I am currently working on the 'Analysis' notebook and have run into the error of :
<img width="1046" alt="image" src="https://github.com/user-attachments/assets/f364d04b-7e47-46c2-a218-1643b33dae31">

I am using Visual Studio on my Mac Book. 
I have used the terminal to install both geoviews and cartopy and then tried to run the code again, but I keep getting the same error. I have confirmed that the packages were installed: 
<img width="1046" alt="image" src="https://github.com/user-attachments/assets/d3028e65-0e94-4679-8718-daa21af08639">

However, when I try to run my import code I still get this for both geoviews and cartopy: 

<img width="1046" alt="image" src="https://github.com/user-attachments/assets/61da2050-e8d9-44bd-8cf9-f5626003a3eb">

Any suggestions as to why the install isn't working. I have also tried the install with both pip and conda.

**Comment by npwelsh** [comment, marked as answer]
Hey I ran into a similar problem this morning, turns out I was using pip in a different environment from what I was running my notebook in.  

To fix, here is what I did:

In the Git Bash Terminal in  VSCode Spaces, I navigated to my python environment directory, in my case:

 `cd ~/miniconda3/envs/earth-analytics-python/
`
I then activated the environment using:  

`conda activate earth-analytics-python`

(note: here I had problem running the above, so I ran `source activate base` first before I ran the above, for whatever reason sometimes this happens to me)

Then I checked to make sure I was in the right environment with pip:

`pip3 --version`

My response was:

`pip 24.2 from C:\Users\Nolan Welsh\miniconda3\envs\earth-analytics-python\Lib\site-packages\pip (python 3.11)
(earth-analytics-python) `

I was able to then you pip to install the library I needed and ran the notebook and it worked!  Let me know if this help!
-Nolan

Sources:
https://stackoverflow.com/questions/62822269/vs-code-not-recognize-the-python-package-i-installed-on-my-mac
https://www.earthdatascience.org/workshops/setup-earth-analytics-python/setup-python-conda-earth-analytics-environment/
https://community.anaconda.cloud/t/unable-to-activate-environment-prompted-to-run-conda-init-before-conda-activate-but-it-doesnt-work/68677

**Comment by eculler** [comment]
@ZimmZone I and others have not had a lot of success installing geoviews with pip. I realize that is often the suggestion Python gives you when a package isn't installed. However, it just is not very good at solving package dependencies, and it's been a problem for years. I recommend using `conda` (ideally making sure that geoviews is in the environment to start with). 

If you're on codespaces, you should be able to rebuild the codespace to get the most recent version of the environment, which has geoviews. Make sure to save and commit your work first! Rebuilding will also delete any data you have downloaded, so you may need to run earlier notebooks again.
